src/porepy/models/solution_strategy.py

```diff
@@ -560,6 +620,10 @@ def assemble_linear_system(self) -> None:
         self.linear_system = self.equation_system.assemble()
         logger.debug(f"Assembled linear system in {time.time() - t_0:.2e} seconds.")
 
+        # Definition of nonlinear solver strategy not finalized - redefine FIXME
+        if not self.nonlinear_solver_statistics.init_complete:
+            self.set_nonlinear_solver_statistics()
```

Comment:
The solver statistics requires info about equation names and variable names. For this it requires an active assembly. This is not elegant but kept for now to start the discussion of the overall PR.